src/Containers/Microsoft.NET.Build.Containers/FallbackToHttpMessageHandler.cs

```diff
@@ -76,10 +79,16 @@ internal static bool ShouldAttemptFallbackToHttp(HttpRequestException exception)
         return exception.HttpRequestError == HttpRequestError.SecureConnectionError;
     }
 
-    private static void FallbackToHttp(HttpRequestMessage request)
+    private static void FallbackToHttp(string registryName, HttpRequestMessage request)
     {
         var uriBuilder = new UriBuilder(request.RequestUri!);
         uriBuilder.Scheme = "http";
+        if (registryName.IndexOf(':') < 0)
```

Comment:
This seems fine, but I wish there was a more structured way to parse 'things that look like domain names'. I'm slightly worried about breaks here.

Reply:
Yes, It not a good way to check this, but as I know, Uri can't see any difference between `https://abc.com:443` and `https://abc.com`, So I have no idea about how to do it better. Do you have any ideas?

Reply:
I've asked the internal brain-trust if they have any guidance for this use case - I'll give them a few hours to respond and if there's no guidance we can go ahead with your current solution.

Reply:
@MihaZupan suggested the following:

> Right now you can get the original port used by using a custom scheme as a workaround.
> Only looking for ':' would actually work fine if you knew that your inputs were host names, but it'd break if they can also be an IPv6 address, which Uri can handle for you.

So I suggest doing this - given the `registryName`, parse it as a uri with a deliberately unknown domain like 'custom' or 'containers', then there will be no default port.

Reply:
Thanks for the suggestion, It's really helpful.

Fixed this and add some unit tests for ipv6 in new commit. Please have a look.